File: shell/src/types/pluginIds.ts
/**
 * Canonical plugin identifiers — keep in sync with
 * `crates/nexus-types/src/plugin_ids.rs`.
 *
 * Every Nexus plugin is referenced by a reverse-DNS string of the form
 * `com.nexus.<name>`. Centralizing the constants here lets shell-side plugin
 * code stop scattering literal strings and stay aligned with Rust.
 *
 * Drift check: a follow-up will emit this file from the Rust source via
 * ts-rs (see Phase 5 P5-02). For now keep entries alphabetized.
 */

export const PLUGIN_IDS = {
  ACP: 'com.nexus.acp',
  AGENT: 'com.nexus.agent',
  AI: 'com.nexus.ai',
  AI_RUNTIME: 'com.nexus.ai.runtime',
  AUDIO: 'com.nexus.audio',
  CLI: 'com.nexus.cli',
  COLLAB: 'com.nexus.collab',
  COMMENTS: 'com.nexus.comments',
  DAP: 'com.nexus.dap',
  DATABASE: 'com.nexus.database',
  EDITOR: 'com.nexus.editor',
  FORMATS: 'com.nexus.formats',
  GIT: 'com.nexus.git',
  KERNEL: 'com.nexus.kernel',
  KV: 'com.nexus.kv',
  LINKPREVIEW: 'com.nexus.linkpreview',
  LSP: 'com.nexus.lsp',
  MCP: 'com.nexus.mcp.host',
  NOTIFICATIONS: 'com.nexus.notifications',
  SECURITY: 'com.nexus.security',
  SKILLS: 'com.nexus.skills',
  STORAGE: 'com.nexus.storage',
  TEMPLATES: 'com.nexus.templates',
  TERMINAL: 'com.nexus.terminal',
  THEME: 'com.nexus.theme',
  TUI: 'com.nexus.tui',
  WORKFLOW: 'com.nexus.workflow',
} as const

export type PluginId = (typeof PLUGIN_IDS)[keyof typeof PLUGIN_IDS]

File: crates/nexus-types/src/plugin_ids.rs
//! Canonical plugin identifiers used across the Nexus tree.
//!
//! Every Nexus plugin is referenced by a reverse-DNS string of the form
//! `com.nexus.<name>`. These constants exist so subsystem crates, frontends,
//! and tests can refer to a plugin without scattering string literals — keeping
//! Rust and the TypeScript shell aligned via ts-rs export.

/// Kernel itself — emits lifecycle / capability events.
pub const KERNEL: &str = "com.nexus.kernel";

/// Storage / file-as-truth subsystem.
pub const STORAGE: &str = "com.nexus.storage";

/// AI plugin — chat, RAG, ask handlers.
pub const AI: &str = "com.nexus.ai";

/// AI runtime — model invocation pipeline.
pub const AI_RUNTIME: &str = "com.nexus.ai.runtime";

/// Agent orchestration plugin.
pub const AGENT: &str = "com.nexus.agent";

/// Comments overlay plugin.
pub const COMMENTS: &str = "com.nexus.comments";

/// Editor plugin (CRDT + transactional file ops).
pub const EDITOR: &str = "com.nexus.editor";

/// Git plugin — branch / dirty / commit surface.
pub const GIT: &str = "com.nexus.git";

/// Link-preview metadata plugin.
pub const LINKPREVIEW: &str = "com.nexus.linkpreview";

/// MCP host (Model Context Protocol bridge).
pub const MCP: &str = "com.nexus.mcp.host";

/// LSP host.
pub const LSP: &str = "com.nexus.lsp";

/// DAP host.
pub const DAP: &str = "com.nexus.dap";

/// ACP host.
pub const ACP: &str = "com.nexus.acp";

/// Skills plugin — authored prompt templates.
pub const SKILLS: &str = "com.nexus.skills";

/// Templates plugin.
pub const TEMPLATES: &str = "com.nexus.templates";

/// Terminal plugin.
pub const TERMINAL: &str = "com.nexus.terminal";

/// Theme plugin.
pub const THEME: &str = "com.nexus.theme";

/// Workflow runner.
pub const WORKFLOW: &str = "com.nexus.workflow";

/// Database plugin.
pub const DATABASE: &str = "com.nexus.database";

/// Key-value store plugin.
pub const KV: &str = "com.nexus.kv";

/// Security / audit plugin.
pub const SECURITY: &str = "com.nexus.security";

/// Format codecs plugin.
pub const FORMATS: &str = "com.nexus.formats";

/// Notifications plugin (inbox + transports).
pub const NOTIFICATIONS: &str = "com.nexus.notifications";

/// Audio plugin.
pub const AUDIO: &str = "com.nexus.audio";

/// Collaboration plugin.
pub const COLLAB: &str = "com.nexus.collab";

/// CLI host (event source for command invocations).
pub const CLI: &str = "com.nexus.cli";

/// TUI host.
pub const TUI: &str = "com.nexus.tui";
